### PR TITLE
Fix error reporting for GetStaticParametersForMethod

### PR DESCRIPTION
**Pull request by dsyme**
This fixes an error reporting problem for the F# 4.0 feature "static parameters for methods" reported by @dmitry-a-morozov  

See https://github.com/fsprojects/FSharp.TypeProviders.StarterPack/issues/63

If possible this should be applied to the next Visual F# Tools update.

### CODE REVIEW

**Comment by dsyme** [comment]
The indication that the fix is good can be found in the correct error messages [here](https://github.com/Microsoft/visualfsharp/pull/703/files#diff-a9445c9d3d798329be92281a5e895a94R2) and [here]()

**Comment by dmitry-a-morozov** [comment]
Cool. Thanks Don.

Would be nice if 
https://github.com/Microsoft/visualfsharp/issues/640
https://github.com/Microsoft/visualfsharp/issues/641
https://github.com/Microsoft/visualfsharp/issues/642

will make into Update 1 too.

I made nice progress on integrating statically parameterized methods with FSharp.Data.SqlClient.
https://github.com/fsprojects/FSharp.Data.SqlClient/blob/V2/src/SqlClient.Tests/CreateCommand.fsx

It's planned for V2. 
But I probably won't ship production ready V2 until these issues resolved because to me they are serious usability blockers.

File: src/fsharp/ExtensionTyping.fs [hunk 1]
```diff
@@ -520,7 +521,9 @@ module internal ExtensionTyping =
                     // To allow a type provider to depend only on FSharp.Core 4.3.0.0, it can alternatively implement an appropriate method called GetStaticParametersForMethod
                     let meth = provider.GetType().GetMethod( "GetStaticParametersForMethod", bindingFlags, null, [| typeof<MethodBase> |], null)  
                     if isNull meth then [| |] else
-                    let paramsAsObj = meth.Invoke(provider, bindingFlags ||| BindingFlags.InvokeMethod, null, [| box x |], null) 
+                    let paramsAsObj = 
+                        try meth.Invoke(provider, bindingFlags ||| BindingFlags.InvokeMethod, null, [| box x |], null) 
+                        with err -> raise (StripException (StripException err))
```
Comment (KevinRansom) [inline src/fsharp/ExtensionTyping.fs]: Is it really possible to double wrap TargetInvocationand TypeInitialization exceptions.  If so shouldn't we just make Stri[Exception recursively strip then until we find a type that is not TInvocation or TypeInitialization ...

Comment (dsyme) [inline src/fsharp/ExtensionTyping.fs]: Hmmm... I'm copying what we did elsewhere for now - perhaps its wrong but it seems ok for here?

**Comment by dsyme** [comment]
This change is good to go in.

**Comment by KevinRansom** [comment]
merged into vs2015

**Comment by dsyme** [comment]
Thanks for merging!

**Pull request closed without merging**